app/test_main.py

```diff
@@ -1 +1,28 @@
-# write your code here
+import pytest
+from unittest.mock import patch
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "mock_valid_google_url_return, "
+    "mock_has_internet_connection_return, "
+    "expected",
+    [
+        (True, True, "Accessible"),
+        (False, True, "Not accessible"),
+        (True, False, "Not accessible"),
+        (False, False, "Not accessible"),
+    ]
+)
+def test_can_access_google_page(
+        mock_valid_google_url_return: bool,
+        mock_has_internet_connection_return: bool,
+        expected: str
+) -> None:
+    with (patch("app.main.valid_google_url") as
+          mock_valid_google_url,
+          patch("app.main.has_internet_connection") as
+          mock_has_internet):
```

Comment:
better to use  `@mock.patch` decorator 
to make your code more readable and clear